interface/AsyncDevice.tt

```diff
@@ -0,0 +1,115 @@
+﻿<#@ template debug="false" hostspecific="true" language="C#" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.IO" #>
+<#@ import namespace="System.Linq" #>
+<#@ import namespace="System.Text" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ include file="Interface.tt" #><##>
+<#@ output extension=".Interface.cs" #>
+<#
+var namespaceName = Host.ResolveAssemblyReference("$(MSBuildProjectName)");
+var firmwarePath = Host.ResolveAssemblyReference("$(HarpFirmwarePath)");
+var metadataPath = !string.IsNullOrEmpty(firmwarePath)
+    ? Host.ResolvePath(firmwarePath)
+    : Path.GetDirectoryName(Host.TemplateFile);
+var metadataFileName = Path.Combine(metadataPath, "device.yml");
+
+var deviceMetadata = TemplateHelper.ReadDeviceMetadata(metadataFileName);
+var publicRegisters = deviceMetadata.Registers.Where(register => register.Value.Visibility == RegisterVisibility.Public).ToList();
+var deviceName = deviceMetadata.Device;
+var deviceClassName = deviceName + "Device";
+#>
+using Bonsai.Harp;
+using System.Threading.Tasks;
+
+namespace <#= namespaceName #>
+{
+    /// <summary>
+    /// Represents an asynchronous API to configure and interface with <#= deviceName #> devices.
+    /// </summary>
+    public partial class <#= deviceClassName #> : AsyncDevice
+    {
+        private <#= deviceClassName #>(string portName)
+            : base(portName)
```

Comment:
Should this not be WhoAmI instead of portName?
In a related note, should we not allow users to override the default WhoAmI. I agree that most of the times it wont matter, but having the option might save someone in the future?

Reply:
But I guess people can always use the generic async device combined with the methods of this device. Ye nvm, disregard my previous comment...